modules/cloudsql-instance/variables.tf

```diff
@@ -169,9 +148,26 @@ variable "name" {
   type        = string
 }
 
-variable "network" {
-  description = "VPC self link where the instances will be deployed. Private Service Networking must be enabled and configured in this VPC."
-  type        = string
+variable "network_config" {
```

Comment:
One more idea here: can we maybe deviate a bit from provider naming (I really don't like `enable_ipv4`) here and have something like that:
```hcl
object({
  authorized_networks = optional(map(string))
  ...
  connectivity = object({
    public_ip  = optional(bool, false)
    psa_config = optional(object({
      private_network     = string
      allocated_ip_ranges = optional(object({
        primary = optional(string)
        replica = optional(string)
      })
    })
    psc_allowed_consumer_projects = optional(list(string))
  })
  ...
})
```

With semantics for `psc_config` that if it is null, then PSC is disabled, if not null, then enabled.

Two non-optional fields above will also make apparent, that user needs to make a choice here and simplify validation a bit.

@juliocc @luigi-bitonti WDYT?

Reply:
I like this. `ipv4_enabled` is really confusing indeed.

Reply:
And it should be `public_ipv4`  instead of `public_ip` in anticipation of IPv6 support.

Reply:
Thank you for the suggestion, I agree with you. I'll made the changes ASAP.

Reply:
Just pushed.